### PR TITLE
feat(datasource/orb): Add support for internal CircleCI Registries

### PR DESCRIPTION
**Pull request by mglazer**
## Changes

Enables CircleCI Orb datasource to use additional registries.

## Context

The current implementation prevented using additional registries, I suspect because most people are not self-hosting
CircleCI. This just enables support for this use case. Note that this is separate from the current [private orb](https://docs.renovatebot.com/modules/manager/circleci/) support which still requires hosting the orbs on the Internet facing CircleCI instance.

This was a pretty simple commit, but took me an inordinately long time to test because I hit a number of extremely weird issues running renovate locally on OSX. Everything seemed to work when I switched to an Ubuntu VM though.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/orb/index.spec.ts [hunk 1]
```diff
@@ -92,5 +92,19 @@ describe('modules/datasource/orb/index', () => {
...
+      httpMock
+        .scope('https://cci.internal.dev')
+        .post('/graphql-unstable')
+        .reply(200, orbData);
+      const res = await getPkgReleases({
+        datasource,
+        packageName: 'hyper-expanse/library-release-workflows',
+        registryUrls: ['https://cci.internal.dev'],
+      });
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/datasource/orb/index.spec.ts:106]: please use to equal or to match object. read our dev guides

File: lib/modules/datasource/orb/index.ts [hunk 2]
```diff
@@ -27,9 +27,10 @@ export class OrbDatasource extends Datasource {
     super(OrbDatasource.id);
   }
 
-  override readonly customRegistrySupport = false;
+  override readonly customRegistrySupport = true;
 
   override readonly defaultRegistryUrls = ['https://circleci.com/'];
+  override readonly registryStrategy = 'merge';
```
Comment (viceice) [inline lib/modules/datasource/orb/index.ts:33]: is it how circle CI works? can a single orb exist on multiple registries?

File: lib/modules/datasource/orb/index.ts [hunk 3]
```diff
@@ -47,7 +48,10 @@ export class OrbDatasource extends Datasource {
     if (!registryUrl) {
       return null;
     }
-    const url = `${registryUrl}graphql-unstable`;
+    const path = registryUrl.endsWith('/')
+      ? 'graphql-unstable'
+      : '/graphql-unstable';
+    const url = `${registryUrl}${path}`;
```
Comment (viceice) [inline lib/modules/datasource/orb/index.ts:54]:
```suggestion
    const url = joinUrlParts(registryUrl, 'graphql-unstable');
```
we've util functions for such things

File: lib/modules/datasource/orb/index.ts (hunk 2 above)
Comment (mglazer) [reply lib/modules/datasource/orb/index.ts:33]: I wasn't quite sure what to pick here. Looks like hunt is a better option based off of my reading.

File: lib/modules/datasource/orb/index.spec.ts (hunk 1 above)
Comment (mglazer) [reply lib/modules/datasource/orb/index.spec.ts]: I was just pattern matching to the rest of this test file. Happy to update it.

File: lib/modules/datasource/orb/index.ts (hunk 3 above)
Comment (mglazer) [reply lib/modules/datasource/orb/index.ts]: Thanks!

**Review by rarkins — DISMISSED** [review]

File: lib/modules/datasource/orb/index.spec.ts [hunk 4]
```diff
@@ -92,5 +92,19 @@ describe('modules/datasource/orb/index', () => {
...
+      httpMock
+        .scope('https://cci.internal.dev')
+        .post('/graphql-unstable')
+        .reply(200, orbData);
+      const res = await getPkgReleases({
+        datasource,
+        packageName: 'hyper-expanse/library-release-workflows',
+        registryUrls: ['https://cci.internal.dev'],
+      });
+      expect(res).not.toBeNull();
```
Comment (viceice) [inline lib/modules/datasource/orb/index.spec.ts:106]:
```suggestion
```

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**